### PR TITLE
Feature  branch - Tic Tac Toe

### PR DESCRIPTION
**Pull request by kiptubei**
# JavaScript-tic-tac-toe-game
This is a Tic Tac Toe game you can play in your browser!

# Project Specification

1. Set up your project with a HTML, CSS and Javascript files and get the Git repo all set up.
2. You’re going to store the gameboard as an array inside of a Gameboard object, so start there! Your players are also going to be stored in objects… and you’re probably going to want an object to control the flow of the game itself.
3. Set up your HTML and write a JavaScript function that will render the contents of the gameboard array to the webpage (for now you can just manually fill in the array with "X"s and "O"s)
4. Build the functions that allow players to add marks to a specific spot on the board, and then tie it to the DOM, letting players click on the gameboard to place their marker. Don’t forget the logic that keeps players from playing in spots that are already taken!
5. Build the logic that checks for when the game is over! Should check for 3-in-a-row and a tie.
6. Clean up the interface to allow players to put in their names, include a button to start/restart the game and add a display element that congratulates the winning player!

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @kiptubei and @tGodson 
### You have done well so far :clap: However, you have a few issues to fix:-
- [x] Don't change the configuration files for linting, revert the changes you made on .eslintrc.json file. [Refer to this link](https://github.com/microverseinc/linters-config/tree/master/javascript)
- [x] Use es6 function to implement your functions, see below
```
const handlePlayerChange = () => {
   # Your code
}
```
- [x] In the code you must-have `game`, `gameBoard`,  `player` objects
- [x] It has a method(s) to check the game's winning conditions
- [x] It allows players to input their name
- [x] It includes a button to reset the game
### Make all these changes and submit for another code review. Good luck.
- Happy coding!

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team
### You have done well so far, a few issues you still need to fix:-
- [x] Do not change the content of .eslintrc.json file, use the one provided by microverse, [check it out here](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json)
- [x] Separate app logic from dom manipulation
### Make these changes and submit for another code review. Good luck!
- Happy coding!

File: script.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+const empty = () => {
+  document.querySelector('#player1').value = '';
+  document.querySelector('#player2').value = '';
+};
+
+const form = document.querySelector('.form');
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  player1 = document.querySelector('#player1').value;
+  player2 = document.querySelector('#player2').value;
```
Comment (rOluochKe) [inline script.js]: ### Separate dom manipulation from app logic

File: script.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+const drawMessage = () => 'Game ended in a draw!';
+
+const currentPlayerName = () => {
+  players = JSON.parse(localStorage.getItem('players'));
+  if (currentPlayer === 'X') {
+    statusDisplay.innerHTML = `It's ${players.player1}'s turn`;
+  } else {
+    statusDisplay.innerHTML = `It's ${players.player2}'s turn`;
+  }
+};
```
Comment (rOluochKe) [inline script.js]: ### Separate dom manipulation from app logic

**Review by adaorachi — APPROVED** [review]
### Hi @kiptubei and @tGodson :wave: 
You've done a good job implementing the changes requested in the previous review. Your code now looks very modular. Well done guys!  That's a thumb's up for me! :+1: 
You can now merge the branch to the base branch of your repo while you start working on the next project.

---

### STATUS: PROJECT APPROVED :100: 

---

Before you merge, there are just a few things I'd like you to look into and which will help you keep better code and structure in this project and future ones.
- [**optional**] You have some loose files in your project structure tree. It will be nice if you keep related files in separate folders. Like the JS files in a JS folder, the stylesheets in its folder, e.t.c. This will help keep your working project directory organized.
- [**optional**] It will be nice if you include a live version of your project in your README. This is very useful for users or even job hirers to get a view and grasp of your project even without seeing your codes.
- [**optional**] I'm sure the use of webpack has been introduced in the Javascript curriculum prior to this project. It will be nice if you made use of it. Webpack is very useful in keeping modular and clean codes and it will also prevent you from including multiple JS files in your HTML file just like you did.

Good luck and Happy Coding :relaxed: 

MaryAnn Chukwuka.

**Pull request merged**